core/src/main/java/org/infinispan/container/entries/RepeatableReadEntry.java

```diff
@@ -17,12 +17,10 @@
    private static final boolean trace = log.isTraceEnabled();
 
    /* Value before the last modification. Serves as the previous value when the operation is retried */
-   protected Object oldValue;
    protected Metadata oldMetadata;
 
    public RepeatableReadEntry(Object key, Object value, Metadata metadata) {
       super(key, value, metadata);
-      this.oldValue = value;
       this.oldMetadata = metadata;
```

Comment:
I think you need to move the `oldMetadata` to `ReadCommittedEntry`

Reply:
moved